record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
It'd be nice if the only clean exit condition was hitting EOF. Remind me why it's bad to try to recover after hitting ErrUnexpectedEOF.